app/controllers/user_sessions_controller.rb

```diff
@@ -50,7 +50,14 @@ def handle_social_login_flow(auth)
         redirect_to return_to + hash_params, notice: "Already linked to another account!"
       end
     else # not signed in
-      if @identity&.user.present?
```

Comment:
Your code should go inside `if @identity&.user.present?`
Does this makes sense?

Reply:
No because the identity does not exist if the user has not already signed in through oauth. If for example, the user has created an account through PL website, logged out, and was then banned. The next day, they try to create a new account through oauth and enter the same email they used for their banned PL account. Without my changes, the user's credentials would go through here: 

https://github.com/publiclab/plots2/blob/d72137f2a52abbd3cd31c0af1ae2a67e43e6a835/app/controllers/user_sessions_controller.rb#L97-L103
The user would be linked to his **banned** PL account and then logged in although he is banned. That's why I can't put the code in the `@identity` section.

Reply:
Do you see what I mean?

Reply:
great!

Reply:
Thanks! I have really though about the signup and login logic a lot since I know how complex this issue is :+1: